rio/src/RIO/Vector/Partial.hs

```diff
@@ -62,3 +63,23 @@ module RIO.Vector.Partial
   ) where
 
 import qualified Data.Vector.Generic
+
+-- | /O(1)/ Yield a slice of the vector without copying it. The vector must
+-- contain at least @i+n@ elements.
+slice :: Data.Vector.Generic.Vector v a
+  => Int   -- ^ @i@ starting index
+  -> Int   -- ^ @n@ length
+  -> v a
+  -> v a
+slice i n v = if i > 0 && n > 0 && i + n < 0  -- `i+n` overflows
```

Comment:
I'd rather see a patch like this land in `vector` than here. I'm also concerned about surprising behavior of error types changing.

Reply:
OK, I will remove 8673eb7c7f07530ae9108ec0490f5e277320bc12 then.